publish.go

```diff
@@ -26,7 +28,11 @@ func NewPublisher(ctx context.Context,
 	if err != nil {
 		return nil, err
 	}
-	return &legPublisher{ss.t, ss.onClose}, nil
+
+	headPublisher := &head.Publisher{}
+	go headPublisher.Serve(ctx, host, topic)
```

Comment:
Passing the ctx into `headPublisher.Serve` does not accomplish anything.  It only makes the `Serve` function stop waiting on its internal goroutine, but does not stop that internal goroutine.  See earlier comment on `Server`.  Also, if then error returned is always ignored, then why return it. Maybe log it, or just don't return it if is not intended to be used.